File: base.net/src/main/java/alvin/base/net/ApplicationModule.java
package alvin.base.net;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
interface ApplicationModule {

    @Singleton
    @Provides
    static Context context(final Application application) {
        return application;
    }
}

File: ui.layout/src/main/java/alvin/ui/layout/ApplicationModule.kt
package alvin.ui.layout

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object ApplicationModule {

    @Provides
    @Singleton
    @JvmStatic
    fun context(application: Application): Context {
        return application
    }
}
